app/models/election.ts

```diff
@@ -73,22 +72,26 @@ const electionSchema = new Schema({
   },
 });
 
-electionSchema.pre('remove', function (next) {
+electionSchema.pre('remove', async function (next) {
   // Use mongoose.model getter to avoid circular dependencies
-  mongoose
+  await mongoose
     .model('Alternative')
     .find({ election: this.id })
-    .then((alternatives) => {
-      Bluebird.map(alternatives, (alternative) => alternative.remove());
-    })
-    .nodeify(next);
-  mongoose
+    .then(async (alternatives) => {
+      await Promise.all(alternatives.map(async alternative => {
+        await alternative.remove();
+      }));
```

Comment:
I feel like there might be some unnecessary async awaits here?
Some of these functions return promises, so you should not need to async and await them, essentially returning a new promise that just waits for another promise? :thinking:

Reply:
I'm pretty sure both the `async (alternatives)` and `async alternative` functions don't need to be async, since the `Promise.all` will return a promise. Same for `alternative.remove`

Reply:
Trueee. I'll remove them for now. I think we'll quickly notice if this not works during testing